test/evergreen/code_coverage/parallel_code_coverage.py

```diff
@@ -0,0 +1,202 @@
+
+        found_compile_time_coverage_files = False
+        found_run_time_coverage_files = False
+
+        # Check build dir for coverage files
+        for root, dirs, files in os.walk(this_build_dir):
+            for filename in files:
+                if filename.endswith('.gcno'):
+                    found_compile_time_coverage_files = True
+                if filename.endswith('.gcda'):
+                    found_run_time_coverage_files = True
+
+        if verbose:
+            print('Found compile time coverage files in {} = {}'.
+                  format(this_build_dir, found_compile_time_coverage_files))
+            print('Found run time coverage files in {}     = {}'.
+                  format(this_build_dir, found_run_time_coverage_files))
+
+        if not setup and not found_compile_time_coverage_files:
+            sys.exit('No compile time coverage files found within {}. Please build for code coverage.'
+                     .format(this_build_dir))
+
+        # if found_run_time_coverage_files:
+        #     sys.exit('Run time coverage files found within {}. Please remove them.'.format(this_build_dir))
+
+    if verbose:
+        print("Build dirs: {}".format(build_dirs))
+
+    setup_bucket_info = []
+    task_bucket_info = []
+    for build_dir in build_dirs:
+        if setup:
+            if len(os.listdir(build_dir)) > 0:
+                sys.exit("Directory {} is not empty".format(build_dir))
+            setup_bucket_info.append({'build_dir': build_dir, 'task_bucket': config['setup_actions'],
+                                      'verbose':verbose})
+        task_bucket_info.append({'build_dir': build_dir, 'task_bucket': [], 'verbose':verbose})
+
+    # Perform setup operations
+    if setup:
+        setup_start_time = datetime.now()
+        with concurrent.futures.ProcessPoolExecutor(max_workers=parallel_tests) as executor:
+            for e in executor.map(run_task_list, setup_bucket_info):
+                print('Setup')
+        setup_end_time = datetime.now()
+        setup_diff = setup_end_time - setup_start_time
+
+        if verbose:
+            print("Time taken to perform the setup: {} seconds".format(setup_diff.total_seconds()))
+
+    # Prepare to run the tasks in the list
+    for test_num in range(len(config['test_tasks'])):
+        test = config['test_tasks'][test_num]
+        build_dir_number = test_num % parallel_tests
+
+        if verbose:
+            print("Prepping test [{}] as build number {}: {} ".format(test_num, build_dir_number, test))
+
+        task_bucket_info[build_dir_number]['task_bucket'].append(test)
+
+    if verbose:
+        print("task_bucket_info: {}".format(task_bucket_info))
+
+    # Perform task operations in parallel across the build directories
+    task_start_time = datetime.now()
+
+    with concurrent.futures.ProcessPoolExecutor(max_workers=parallel_tests) as executor:
+        for e in executor.map(run_task_list, task_bucket_info):
+            print('Test')
```

Comment:
Is `for e` required here so the tests execute?
If so we can replace this print with 
```suggestion
            # We don't need to check the results of each test as we're only generating coverage data
            pass
```

Reply:
In this case, `e` is the result of the operation. I've changed the function to return a string (which will end up in 'e') and is now printed out.